lib/gateways/snapshot-gateway.js

```diff
@@ -66,15 +66,23 @@ class SnapshotGateway {
       if (!s.systemIds || s.systemIds.length === 0) return true;
       return s.systemIds.some(id => systemIds.includes(id));
     };
-    const matches = await Promise.all(Items.filter(matchBySystemId));
+    const matches = Items.filter(matchBySystemId);
 
     // because we're using a GSI to find, we need to then do a get
     // on each match to get the full data model back
-    const snapshots = (
-      await Promise.all(matches.map(m => this.get(m)))
-    ).sort((s1, s2) => (s1.created < s2.created ? 1 : -1));
+    const snapshots = await Promise.all(matches.map(m => this.get(m)));
 
-    return { snapshots };
+    const filterEmptySnapshots = s =>
+      s.vulnerabilities.length > 0 || s.assets.length > 0 || s.notes.length > 0;
+
+    const sortSnapshotsByCreateDate = (s1, s2) =>
+      s1.created < s2.created ? 1 : -1;
```

Comment:
If these are dates an alternative could be `s1.created.valueOf() - s2.created.valueOf()`, but this way is probably easier to read. It doesn't handle the case where dates are equal - but that seems extremely unlikely given that they are probably stored to millisecond precision.

Reply:
If they're equal then it'll return -1 but it doesn't matter which of the two go first